src/sql/parts/dashboard/containers/dashboardNavSection.contribution.ts

```diff
@@ -60,9 +63,43 @@ let NavSectionSchema: IJSONSchema = {
 
 registerContainerType(NAV_SECTION, NavSectionSchema);
 
-export function validateNavSectionContribution(extension: IExtensionPointUser<any>, navSectionConfigs: NavSectionConfig[]): boolean {
+function isValidIcon(icon: IUserFriendlyIcon, extension: IExtensionPointUser<any>): boolean {
+	if (typeof icon === 'undefined') {
+		return false;
+	}
+	if (typeof icon === 'string') {
+		return true;
+	} else if (typeof icon.dark === 'string' && typeof icon.light === 'string') {
+		return true;
+	}
+	extension.collector.error(nls.localize('opticon', "property `icon` can be omitted or must be either a string or a literal like `{dark, light}`"));
+	return false;
+}
+
+function createCSSRuleForIcon(icon: IUserFriendlyIcon, id: string, extension: IExtensionPointUser<any>): void {
+	let iconClass: string;
+	if (icon) {
+		iconClass = id;
+		if (typeof icon === 'string') {
+			const path = join(extension.description.extensionFolderPath, icon);
```

Comment:
vscode generates random guid for extension contributed icons, probably to reduce the chance of conflicts

Reply:
I will add that